src/common/transformations/include/transformations/utils/utils.hpp

```diff
@@ -230,6 +232,28 @@ TRANSFORMATIONS_API bool is_constant_and_all_values_equal_int(const Output<Node>
 
 TRANSFORMATIONS_API bool is_on_constant_path(const ov::Output<ov::Node>& output);
 
+template <typename T>
+ov::pass::pattern::op::ValuePredicate constant_predicate(std::function<bool(const std::vector<T>&)> predicate) {
+    return pass::pattern::op::as_value_predicate([=](std::shared_ptr<Node> n) -> bool {
+        if (auto constant = as_type_ptr<v0::Constant>(n)) {
+            auto values = constant->cast_vector<T>();
+            return predicate(values);
+        }
+        return false;
+    });
+}
 }  // namespace util
 }  // namespace op
 }  // namespace ov
+
+#define INT_CONSTANT_WITH_PREDICATE(expression)                                                   \
+    pattern::wrap_type<op::v0::Constant>(                                                         \
+        ov::op::util::constant_predicate<int64_t>([](const std::vector<int64_t>& value) -> bool { \
+            return expression;                                                                    \
+        }))
+
+#define FLOAT_CONSTANT_WITH_PREDICATE(expression)                                             \
+    pattern::wrap_type<op::v0::Constant>(                                                     \
+        ov::op::util::constant_predicate<float>([](const std::vector<float>& value) -> bool { \
+            return expression;                                                                \
+        }))
```

Comment:
Why macro? Can it be a function that takes a lambda with a single parameter (for value)?

Reply:
to make pattern more distilled. you can see the usage in the patterns down the code.